generator/src/generator/namespace.rs

```diff
@@ -1976,6 +1976,58 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                 );
             }
         }
+
+        // Generate accessors for length fields that are not part of the struct.
+        // First, collect all relevant fields.
+        let deducible = fields
+            .iter()
+            .filter_map(|field| match field {
+                xcbdefs::FieldDef::Normal(normal_field) => deducible_fields
+                    .get(&normal_field.name)
+                    .and_then(|d| match d {
+                        DeducibleField::LengthOf(list_name, op) => Some((list_name, op)),
+                        // switches are not yet supported
+                        _ => None,
+                    })
+                    .map(|d| (field, d)),
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+        if !deducible.is_empty() {
+            outln!(out, "impl {} {{", name);
```

Comment:
The clippy warning can be fixed with something like this
```rust
if deducible.iter().any(|(name, _)| name == Some("len")) {
    outln!(out, "#[allow(clippy::len_without_is_empty)]");
}
```

Reply:
Urgh. I was trying to be clever and have this directly at the `fn`, but it really needs to be at the `impl`.